src/main/scala/tech/sourced/gemini/ReportSparkApp.scala

```diff
@@ -1,34 +1,49 @@
 package tech.sourced.gemini
 
+import com.datastax.driver.core.Cluster
+import com.datastax.driver.core.Session
+
 object ReportSparkApp extends App {
   def printUsage(): Unit = {
-    println("Usage: ./report <repository>")
+    println("Usage: ./report [--detailed]")
     println("")
-    println("Finds duplicated files among hashed repositories")
-    println("  <repository> - repository url, example: github.com/src-d/go-git.git")
+    println("Finds duplicate files among hashed repositories")
+    println("  --detailed to see file paths of each duplicated file")
     System.exit(2)
   }
 
-  if (args.length <= 0) {
-    printUsage()
+  def printIsEmpty[T](report:Iterable[T]): Unit = {
+    if (report.isEmpty) {
+      println(s"No duplicates found.")
+    }
   }
 
-  val repository = args(0)
-  println(s"Reporting all duplicates of: $repository")
+  var detailed = false
+  if (args.length > 0) {
+    val arg = args(0)
+    if (arg == "--detailed") {
+      detailed = true
+    } else {
+      printUsage()
+    }
+  }
 
-  val similar = Gemini.report(repository)
+  val cluster = Cluster.builder().addContactPoint(Gemini.defaultCassandraHost).build()
+  val session = cluster.connect()
 
-  //for every file
-  // query Cassandry by hash
-  // .collect()
-  //print
+  if (detailed) {
+    val report = Gemini.detailedReport(session)
+    printIsEmpty(report)
+    report.foreach( duplicateFiles => {
+      val count = duplicateFiles.count(_=>true)
+      println (s"$count duplicates:\n\t" + (duplicateFiles mkString ("\n\t")))
+    })
+  } else {
+    val report = Gemini.report(session)
+    printIsEmpty(report)
```

Comment:
These two lines look very similar in both cases of `if`, so it may be it could better to move them out of `if` with a flag like  `Gemini.report(session, detailed)` and keep `if` here only for conditional result display?